io-engine/src/core/block_device.rs

```diff
@@ -212,7 +223,51 @@ pub trait BlockDeviceHandle {
         cb_arg: IoCompletionCallbackArg,
     ) -> Result<(), CoreError>;
 
+    /// TODO
```

Comment:
Please add meaningful comments for all new methods, ideally with args and return value descriptions.